### ISSUE TITLE
Gamerule for hiding duplicate player names in player lists.

### ISSUE THREAD

**Issue by PsychoBreadstick** [Ladysnake/Impersonate#27]
A friend and I are planning to use impersonate for some multiplayer shenanigans in the near future, but the feature that shows the new player's username in the player list on servers kind of gives away that there's someone disguised as a certain player when the player you're impersonating joins. Showing two of the same player on the player list. 

Therefore, I propose some sort of solution: A gamerule that hides the impersonating player from the player list **if** the true player that is being impersonated by someone else joins and is online. For example:

Someone impersonates the Notch account. And on the tab player list, it shows Notch just normally. But when the real Notch account joins, it won't show two Notches on the player list, just one.

**Comment by Pyrofab** [comment]
For technical reasons, this cannot be done serverside, so it would require players to install impersonate clientside when the gamerule is enabled for it to work as intended. Also I guess you would want duplicated join and leave messages to be removed too with that gamerule ?

**Comment by PsychoBreadstick** [comment]
For the duplicated join and leave messages, yeah they'll likely have to be disabled when the gamerule is active or just completely removed. Also for the issue with it not being done purely serverside. I don't really have a problem with that, but I'm not sure about everyone else who would use this mod for their servers. Although to be fair, if a server is centered around being modded, people are likely going to install the mod clientside anyways. Unless the server's primary focus is to be playable on vanilla clients while having the mod installed serverside.

**Comment by PsychoBreadstick** [comment]
Of course, a solution to the problem is to make the gamerule simply prevent players without the mod installed on their clients from joining the server. Although I'm not sure how well that would work with the gamerule changing in the server while players without the mod are playing, as there could be some odd issues with that. So with a solution like that, the gamerule might need to become a config option instead to prevent odd issues like that, since config options only take effect once the server starts up. Likely preventing any potential issues with the option being toggled on while players without the mod on their client are online.
